FixMyBerlin/fixmy.frontend
# Fix/#34 location access denial handling

# Description

Extends the LocatorControl:
* when the user denies the request to access her geolocation, a custom `ErrorMessage` component is shown instead of an alert
* refactoring (use functional component with hooks)
Fixes https://github.com/FixMyBerlin/fixmy.platform/issues/34

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

- [ ] Test A: Usage in reports - OverviewMap

* go to http://localhost:8080/meldungen/radbuegel/friedrichshain-kreuzberg/karte
* If you have already permitted FixMyBerlin@localhost:8080 to use your geolocation, revoke that permission. As an example, see [how this is done in chrome](https://superuser.com/a/591763)
* use the LocatorControl, allow GeoLocation access
* revoke the permission again
* use the LocatorControl, this time **do not** allow GeoLocation access
* inspect the user feedback provided by the `ErrorMessage` component, and that the `LocatorControl` is disabled for the rest of the session

- [ ] Test B: Usage in /planungen

* do the steps from _Test A_ in http://localhost:8080/planungen

# Implementation Notes / Open aspects

Two things are not optimal, but I can live with them:

1. The `LocatorControl` spawns its own `ErrorMessage`

We should make sure there is only one ErrorMessage instance by moving the component and the `ErrorState` to the top level, see related issues https://github.com/FixMyBerlin/fixmy.platform/issues/23 and https://github.com/FixMyBerlin/fixmy.platform/issues/57.

2. When the user denies the location in /reports and goes to /planungen, the `LocatorButton` is not disabled (and vice-versa).

This is because two different component instances are used and the information about the denial is not persisted in Redux State.

If this bugs us, we should move this state piece to our store.

# Checklist:
